app/lib/foreman_deployments/config/array.rb

```diff
@@ -0,0 +1,83 @@
+module ForemanDeployments
+  module Config
+    class Array < ::Array
+      def configure(values)
+        values = ensure_hash(values)
+        values.map do |index, value|
+          configure_index(index, value)
+        end
+      end
+
+      def configured
+        map.with_index do |item, index|
+          if item.respond_to?(:configured)
+            item.configured
+          elsif item.nil?
+            configuration_storage[index]
+          else
+            item
+          end
+        end
+      end
+
+      def configuration
+        result = configuration_storage.deep_clone
+        each_with_index do |item, index|
+          if item.respond_to?(:configuration)
+            result[index] = item.configuration unless item.configuration.nil?
+          end
+        end
+        result.empty? ? nil : result
+      end
+
+      def transform!(&block)
+        self.map! do |item|
+          block.call(item)
+        end
+        self
+      end
+
+      protected
+
+      def configuration_storage
+        @configuration_storage ||= {}
+      end
+
+      def ensure_hash(values)
+        if values.is_a? ::Hash
+          ::Hash[values.map { |index, value| sanitize_pair(index, value) }]
+        elsif values.is_a? ::Array
+          ::Hash[values.map.with_index { |value, index| sanitize_pair(index, value) }]
+        else
+          fail("Unexpected type #{values.class}")
```

Comment:
`#ensure_hash` checks the validity of a configuration hash that comes from the client. Basically this error should propagate to the user, so it needs to be translated.